### PR TITLE
Movement

### PR DESCRIPTION
**Pull request by pyarosh**

### CODE REVIEW

**Review by pyarosh — COMMENTED** [review]
Only made changes to Model (Tests, Pieces, Move, Chess, Board).
Removed ChessBoard class
Made slight Refactor in GameVerticle to accommodate Chess instead of ChessBoard
Passes all Model Tests

**Comment by nicoburniske** [comment]
I will review it shortly

File: src/main/java/com/burnyarosh/board/common/Coord.java [hunk 1]
```diff
@@ -62,15 +62,28 @@ public String toString() {
         return res;
     }
 
+    public String toChessString(){
```
Comment (nicoburniske) [inline src/main/java/com/burnyarosh/board/common/Coord.java:65]: I'd probably just make this toString.

File: src/main/java/com/burnyarosh/board/common/Coord.java [hunk 2]
```diff
@@ -62,15 +62,28 @@ public String toString() {
...
 
+    @Override
+    public boolean equals(Object o){
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Coord coord = (Coord) o;
+        return this.x == coord.getX() && this.y == coord.getY();
+    }
+    private String getRank(){
+        return ""+(this.y + 1);
```
Comment (nicoburniske) [inline src/main/java/com/burnyarosh/board/common/Coord.java]: Integer.toString or String.valueOf. This appending is kinda weird.

File: src/main/java/com/burnyarosh/board/common/Move.java [hunk 3]
```diff
@@ -1,111 +1,177 @@
...
-            } else {
-                sb.append("0-0-0");
+    /**
+     *
+     * @param b
+     * @param origin
+     * @param target
+     * @return
+     */
+    public static Type classifyMove(Board b, Coord origin, Coord target){
```
Comment (nicoburniske) [inline src/main/java/com/burnyarosh/board/common/Move.java:85]: I like this. Good work

File: src/main/java/com/burnyarosh/board/common/Coord.java [hunk 4]
```diff
@@ -62,15 +62,28 @@ public String toString() {
...
+    }
+
     private String getFile() {
-        String[] file = new String[]{"a", "b", "c", "d", "e", "f", "g", "h"};
-        return file[this.x];
+        return ""+(char) (97 + this.x);
```
Comment (nicoburniske) [inline src/main/java/com/burnyarosh/board/common/Coord.java]: String.valueOf here too

File: src/main/java/com/burnyarosh/board/common/Move.java [hunk 5]
```diff
@@ -1,111 +1,177 @@
...
+        return Type.STANDARD;
+    }
+
+    /**
+     *
+     * @param b
+     * @param promo
+     * @return
+     */
+    private String toAlgebraicNotation(Board b, char promo){
```
Comment (nicoburniske) [inline src/main/java/com/burnyarosh/board/common/Move.java:112]: Great work here.

**Review by nicoburniske — CHANGES_REQUESTED** [review]
This is a good starting point. We still need to agree on a json spec for the board state. Also we should consider adding some custom exceptions for invalid moves and such. basic cases that I can think of are, other players move, and invalid move. We can discuss this more in person tn.

If you could just make the small changes requested I'm ready to merge this into master. Also add some brief documentation when you get that chance. I feel like the naming makes things fairly clear already but each method should have purpose statement. the parameter javadoc is largely unnecessary for private methods. Public methods should have them however.

File: src/main/java/com/burnyarosh/board/common/Coord.java (hunk 1 above)
Comment (nicoburniske) [reply src/main/java/com/burnyarosh/board/common/Coord.java:65]: On second thought this seems fine.

**Review by pyarosh — COMMENTED** [review]
Fixed the awkward String returns in Coord

**Pull request merged**